geode-core/src/integrationTest/java/org/apache/geode/cache/query/internal/index/CompactRangeIndexJUnitTest.java

```diff
@@ -442,6 +445,48 @@ public void doTestHook(final SPOTS spot, final DefaultQuery _ignored,
     }
   }
 
+  /**
+   * Tests adding entries to compact range index where there are undefined and null values
+   * for the key
+   */
+  @Test
+  public void testNullAndUndefinedValuesForMapKeyInCompactRangeIndex() throws Exception {
+    index = utils.createIndex("indexName", "positions['SUN']", SEPARATOR + "exampleRegion");
+    Region<Object, Object> region = utils.getCache().getRegion("exampleRegion");
+
+    // create objects
+    Portfolio p1 = new Portfolio(1);
+    p1.positions = new HashMap<>();
+    p1.positions.put("SUN", "yes");
+    region.put("KEY-" + 1, p1);
+
+    // null value for positions['SUN']
```

Comment:
This comment (and some others in this test) are inaccurate

Reply:
ok